src/power_grid_model/_utils.py

```diff
@@ -143,7 +145,7 @@ def get_batch_size(batch_data: BatchArray) -> int:
     return n_batches
 
 
-def split_numpy_array_in_batches(data: np.ndarray, component: str) -> List[np.ndarray]:
+def split_numpy_array_in_batches(data: np.ndarray, component: ComponentType) -> List[np.ndarray]:
```

Comment:
I believe there is a data type to go for the first parameter here, instead of plainly `np.ndarray`

Reply:
@Jerry-Jinfeng-Guo It would be of the type `Union[DenseBatchArray, SingleArray]`. Although the former is shorter, it probably makes sense to use the existing custom datatypes as you are suggesting.